### PR TITLE
clio-25384 fix bug in pagination code - use next_offset returned by A…

### PR DESCRIPTION
**Pull request by PANCAKES**
…PI call

### CODE REVIEW

File: src/Zap.coffee [hunk 1]
```diff
@@ -60,23 +60,21 @@ Zap.flatten_array = (array, default_keys) ->
 
 Zap.custom_field_definitions = {}
 Zap.custom_field_association_id_postfix = " Id"
-Zap.custom_field_fetch_limit = 200;
+Zap.custom_field_fetch_limit = 5;
```
Comment (tvongaza) [inline src/Zap.coffee]: Obviously change post QA

File: src/Zap.coffee [hunk 2]
```diff
@@ -60,23 +60,21 @@ Zap.flatten_array = (array, default_keys) ->
...
-  while next_offset == max_fetched
+  loop
     url = "custom_fields?offset=#{next_offset}&limit=#{limit}"
     url += "&parent_type=#{parent_type}" if parent_type?
     response_data = Zap.make_get_request(bundle,url)
     custom_fields.push.apply(custom_fields,response_data.custom_fields)
-    next_offset += response_data.records
-    max_fetched += limit
+    next_offset = response_data.next_offset
+    break unless response_data.records == response_data.limit
```
Comment (JefCurtis) [inline src/Zap.coffee]: If I'm understanding the docs correctly shouldn't this be `break unless response_data.records <= response_data.limit`. If we have less records than the limit then there is no reason for the next request?

File: src/Zap.coffee (hunk 1 above)
Comment (PANCAKES) [inline src/Zap.coffee]: Aye

File: src/Zap.coffee (hunk 2 above)
Comment (PANCAKES) [inline src/Zap.coffee]: Good catch!

Comment (PANCAKES) [inline src/Zap.coffee]: I think I am correct here. If the records == the limit, we want to check the next page (continue in the loop -> NOT break)

**Pull request merged**